misc/nsis/install.nsh

```diff
@@ -0,0 +1,547 @@
+        ${Default} ; all other error codes - uninstaller could not start, elevate, etc. - Abort installer
+          SetErrorLevel $0
+          BringToFront
+          Abort "Error executing uninstaller."
+      ${EndSwitch}
+    ${endif}
+
+    ; the uninstaller doesn't delete itself when not copied to the temp directory
+    !insertmacro DeleteRetryAbort "$3\${UNINSTALL_FILENAME}"
+    RMDir "$3"
+  ${endif}
+
+  ; Remove the old uninstaller if it's leftover
+  IfFileExists "$INSTDIR\uninst.exe" 0 +2
+  Delete "$INSTDIR\uninst.exe"
+
+  SetOutPath $INSTDIR
+  ; Write uninstaller and registry uninstall info as the first step,
+  ; so that the user has the option to run the uninstaller if something goes wrong
+  WriteUninstaller "${UNINSTALL_FILENAME}"
+  ; or this if you're using signing:
+  ; File "${UNINSTALL_FILENAME}"
+  !insertmacro MULTIUSER_RegistryAddInstallInfo ; add registry keys
+  ${if} ${silent} ; MUI doesn't write language in silent mode
+    WriteRegStr "${MUI_LANGDLL_REGISTRY_ROOT}" "${MUI_LANGDLL_REGISTRY_KEY}" \
+      "${MUI_LANGDLL_REGISTRY_VALUENAME}" $LANGUAGE
+  ${endif}
+
+  File /r "${DIST_DIR}\*.*"
+SectionEnd
+
+SectionGroup /e "System Integration" SectionGroupIntegration
+
+Section "Register with Windows" SectionWindowsRegister
+  SectionIn 1 2
+
+  ; No HKCU support for Windows versions earlier than Win8
+  ${if} $MultiUser.InstallMode == "AllUsers"
+  ${orif} ${AtLeastWin8}
+    ;StartMenuInternet
+    StrCpy $0 "$INSTDIR\${PROGEXE}"
+    System::Call 'kernel32::GetLongPathNameW(t r0, t .r1, i ${NSIS_MAX_STRLEN}) i .r2'
+
+    StrCpy $0 "SOFTWARE\Clients\StartMenuInternet\${PRODUCT_NAME}"
+
+    !insertmacro UpdateRegStr SHCTX "$0" "" "${PRODUCT_NAME}"
+
+    !insertmacro UpdateRegStr SHCTX "$0\DefaultIcon" "" "$1,0"
+
+    !insertmacro UpdateRegDWORD SHCTX "$0\InstallInfo" "IconsVisible" 1
+
+    !insertmacro UpdateRegStr SHCTX "$0\shell\open\command" "" "$\"$1$\""
+
+    !insertmacro UpdateRegStr SHCTX "$0\shell\properties" "" "${PRODUCT_NAME} settings"
+    !insertmacro UpdateRegStr SHCTX "$0\shell\properties\command" "" "$\"$1$\" ${SHELL_PROPERTIES}"
+
+    !insertmacro UpdateRegStr SHCTX "$0\shell\safemode" "" "${PRODUCT_NAME} safe mode"
+    !insertmacro UpdateRegStr SHCTX "$0\shell\safemode\command" "" "$\"$1$\" ${SHELL_SAFEMODE}"
+
+    !insertmacro UpdateRegStr SHCTX "$0\Capabilities" "ApplicationDescription" "${COMMENTS}"
+    !insertmacro UpdateRegStr SHCTX "$0\Capabilities" "ApplicationIcon" "$1,0"
+    !insertmacro UpdateRegStr SHCTX "$0\Capabilities" "ApplicationName" "${PRODUCT_NAME}"
+
+    !insertmacro UpdateRegStr SHCTX "$0\Capabilities\FileAssociations" ".htm" "${PRODUCT_NAME}HTML"
+    !insertmacro UpdateRegStr SHCTX "$0\Capabilities\FileAssociations" ".html" "${PRODUCT_NAME}HTML"
+    !insertmacro UpdateRegStr SHCTX "$0\Capabilities\FileAssociations" ".shtml" "${PRODUCT_NAME}HTML"
+    !insertmacro UpdateRegStr SHCTX "$0\Capabilities\FileAssociations" ".svg" "${PRODUCT_NAME}HTML"
```

Comment:
No, the default browser just changes the htm,html extensions and http,https protocols. This just tells Windows which filetypes the program can handle. It will be available in the Explorer's "Open with" list of programs for the selected type. In fact, now that .pdf will be supported, it should also be added.

Reply:
Ah, I see! Yeah, sounds like a good idea to add `.pdf`.

Reply:
Added in 4afa2f8.